source/PlayerInfo.cpp

```diff
@@ -776,7 +776,7 @@ map<const shared_ptr<Ship>, vector<string>> PlayerInfo::FlightCheck() const
 	
 	auto flightChecks = map<const shared_ptr<Ship>, vector<string>>{};
 	for(const auto &ship : ships)
-		if(ship->GetSystem() == system && !ship->IsDisabled() && !ship->IsParked())
+		if(ship->GetSystem() && !ship->IsDisabled() && !ship->IsParked())
```

Comment:
I think that ships unable to launch in other systems should be treated slightly differently.
Maybe just add a check later on if the ship is in the same system and if not specify which system the ship is from?